docs/pre-commit.md

```diff
@@ -0,0 +1,33 @@
+# Using pre-commit locally
+
+### Install and setup pre-commit
+- Install:
+  - Using pip: `pip install pre-commit`
+  - Using Homebrew: `brew install pre-commit`
+
+- Setup
+  - run `pre-commit install` to set up the git hook script
+
+### Things to keep in mind regarding the git authoring lifecycle (HEAD, Staged, Unstaged) and pre-commit hooks
+
+Running hooks on unstaged changes can lead to both false-positives and false-negatives during committing. pre-commit only runs on the staged contents of files by temporarily saving the contents of your files at commit time and stashing the unstaged changes while running hooks.
+
+> pre-commit itself will never touch the staging area. These are good ways to silently break commits. In my mind this is one of the worst things that lint-staged does and suggests -- hooks are very frequently not perfect and magically changing what's being committed should not be taken lightly.
+
+Always stage the the changes you intend to commit before running `pre-commit`. If a hook changes our code and fails you will need to manually stage the changes made by `pre-commit`, viewing them one by one with `git add -p` or your preferred git GUI.
```

Comment:
What does "before running `pre-commit`" mean? Running it with the bash script below? (Once I've done `pre-commit install`, it runs automatically every time I try to commit, so everything is always already staged.)

Reply:
means anytime pre-commit hooks are run, whether running pre-commit explicitly or by committing. but ill clarify it